.github/workflows/ci-pipeline.yml

```diff
@@ -62,4 +62,26 @@ jobs:
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
           SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
         if: ${{ always() }}
+  publish:
+    name: Publish to Docker Hub and deploy to Azure
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.ref == 'refs/heads/main' && github.event_name == 'push' && github.event.schedule != '08 13 * * *'
```

Comment:
Given that you've asked this to run on push, do you need to specify that it's also _not_ a schedule? That feels inherently true to me

Reply:
I initially had it set up without any conditions so it ran each time. I then tested it on a test branch:

`github.ref == 'refs/heads/main-test' && github.event_name == 'push' && github.event.schedule != '08 13 * * *'`

It didn't run when I opened a merge request; it only ran when I actually merged the branch. I then changed the target branch to main. 
![image](https://github.com/CameronB9/DevOps-Course-Starter/assets/135842865/6f57a054-9cb4-417e-9424-7bfa49ceb3ae)



> Given that you've asked this to run on push, do you need to specify that it's also not a schedule? That feels inherently true to me

I will remove it, I thought it wasn't needed but decided to leave it in anyway.